doc/doxygen/headers/fe_vs_mapping_vs_fevalues.h

```diff
@@ -0,0 +1,333 @@
+ * points, mappings do not in general need to provide the ability to
+ * at <i>arbitrary</i> points. Rather, as we will see below, they will
+ * be initialized to use a set of quadrature points defined on the
+ * reference cell, will then be "re-initialized" for a particular cell,
+ * and all further operations will then only require
+ * the evaluation of $\mathbf F_K$ at these quadrature points on the
+ * real cell.
+ *
+ * The mapping classes then have the dual role to (i) compute geometric
+ * information (e.g., the normal vectors, determinants of the Jacobians, etc)
+ * and putting them into the data structures from which FEValues can
+ * provide them to the user, and (ii) to provide the support finite
+ * elements need to map shape functions and their derivatives from
+ * the reference cell to the real cell.
+ *
+ *
+ * <h3>Finite elements</h3>
+ *
+ * Finite element classes (i.e., classes derived from FiniteElement) are
+ * responsible for defining their shape functions, derivatives, and many
+ * other aspects on the reference cell, but also for computing the mapped
+ * values and derivatives on actual cells (obviously with the help of a
+ * mapping object). For the current discussion, only the latter role is
+ * important.
+ * 
+ * As with mappings, all that is important for us here is that the finite
+ * element classes can provide this information at given quadrature points,
+ * and that they can put the computed information into structures provided
+ * by FEValues and from which FEValues member functions can then pass
+ * it on to the user through the member functions in FEValuesBase.
+ *
+ *
+ * <h2>What to compute?</h2>
+ *
+ * Let's say a user wants to compute the gradients of shape functions,
+ * for example to compute the integral above. Then she would initialize
+ * an FEValues object by giving the update_gradients flag (as is done
+ * in basically every tutorial program, starting with step-3). What
+ * this indicates is that the user expects the FEValues object to be
+ * able to provide the gradients of shape functions on the real cell,
+ * but expressed no expectation of any other information.
+ *
+ * FEValues will then first have to find out what the mapping and
+ * finite element objects actually require of each other to make this happen.
+ * This already happens at the time the FEValues constructor is run.
+ * Because the mapping does not depend on the finite element (though the
+ * latter does depend on the former), FEValues first asks the finite
+ * element via FiniteElement::requires_update_flags() which <i>other</i>
+ * pieces of information it also requires to make the user request
+ * happen. As an example, if the finite element were of type
+ * FE_Q, then it would determine that in order to compute the
+ * gradients of the shape functions on the real cell $K$, it will
+ * need to compute the gradients of the shape functions on the
+ * reference cell (something it can affect itself, without any
```

Comment:
'affect itself' sounds strange (but I'm not a native English speaker). Maybe 'do on its own' instead?